app/assets/src/components/views/SampleViewV2/ReportTable.jsx

```diff
@@ -354,10 +390,38 @@ class ReportTable extends React.Component {
   };
 
   toggleExpandGenus = ({ taxId }) => {
-    const { expandedGenusIds } = this.state;
+    const { data } = this.props;
+    const { expandedGenusIds, expandAllOpened } = this.state;
     expandedGenusIds.delete(taxId) || expandedGenusIds.add(taxId);
 
-    this.setState({ expandedGenusIds: new Set(expandedGenusIds) });
+    let newExpandedAllOpened = expandAllOpened;
+    if (expandedGenusIds.size === data.length) {
+      newExpandedAllOpened = true;
+    } else if (!expandedGenusIds.size) {
+      newExpandedAllOpened = false;
+    }
+
+    this.setState({
+      expandedGenusIds: new Set(expandedGenusIds),
+      expandAllOpened: newExpandedAllOpened,
+    });
+  };
+
+  toggleExpandAll = () => {
+    const { data } = this.props;
+    const { expandAllOpened } = this.state;
+
+    if (expandAllOpened) {
```

Comment:
Could `expandedGenusIds` be sorted? Currently if you have the table sorted by a column, you lose that sorting when toggling expand all rows.

(Alternatively I suppose a user could expand/close all rows first and then sort. 😛)

Reply:
Fixed.